.changeset/biome_deserialize_context

```diff
@@ -0,0 +1,54 @@
+---
+biome_deserialize: major
+biome_deserialize_macros: major
+---
+
+## Replace `diagnostics` with `context`
+
+All deserialization methods take a `name` parameter.
+This parameter is used to name the deserialized value.
+For instance, in the case of deserialized value from an object field, the name is set to the field name.
+However, this parameter has been abused to pass data such as file names between deserializers.
+
+When we introduced macros to automatically implement `Deserializable`, users request a way of passing the name parameter between deserializers.
+Then, we added the `passthrough_name` attribute to satisfy the request.
+As said before, this was never intended for this usage.
+There are some inherent limitations, notably some deserializers such as the deserializers of `Vec` set `name` to the empty string for all deserialized items.
+
+We now introduce **deserialization context** to provide a proper way to pass a filename or any string identifier to all deserializers.
+All deserializers now take a context parameter that stores this identifier.
+We take the opportunity to integrate diagnostic reporting directly in this deserialization context.
+
+This is a **breaking change**.
```

Comment:
The breaking change paragraph part should be the first paragraph, so it catches the attention of the reader immediately